src/ObjectMultiplex.ts

```diff
@@ -0,0 +1,93 @@
+import { Duplex } from 'readable-stream';
+import eos from 'end-of-stream';
+import once from 'once';
+import { Substream } from './Substream';
+
+const IGNORE_SUBSTREAM: Record<string, unknown> = {};
+
+export class ObjectMultiplex extends Duplex {
+
+  private _substreams: Record<string, Substream | Record<string, any>>;
+
+  constructor(_opts: Record<string, unknown> = {}) {
+    const opts = Object.assign({}, _opts, {
+      objectMode: true,
+    });
+    super(opts);
```

Comment:
This can be simplified, and use the spread operator:
We should use the spread operator (`...`) here instead:
```suggestion
  constructor(opts: Record<string, unknown> = {}) {
    super({
      ...opts,
      objectMode: true,
    });
```